src/coreclr/vm/ceemain.cpp

```diff
@@ -2001,6 +2001,12 @@ static void InitializeDebugger(void)
         }
     }
 
+    static ConfigDWORD debuggerDisablesCodeVersioning;
+    if( (debuggerDisablesCodeVersioning.val(CLRConfig::EXTERNAL_DebuggerLaunchDisablesCodeVersioning) != 0) && CORDebuggerAttached())
+    {
+        LOG((LF_CORDB, LL_INFO10, "Debugger is active at startup, disabling code versioning to prevent a potential deadlock.\n"));
+        g_pConfig->DisableDefaultCodeVersioning();
```

Comment:
Is this guaranteed to happen before the other affected config settings are accessed? From a quick scan it looks like at least one of the tiering config settings is accessed fairly early during startup, so just wanted to make sure this change would be reflected early enough.

Reply:
Is there a particular one you are worried about? When I looked it all seemed ok, but I may have missed something.

The debugger is initialized fairly early in the process, it's after initializing the main subsystems but before we do anything that messes with managed code.

Reply:
Ok sounds good. There seems to be an access to `EEConfig::TieredCompilation()` during `SystemDomain::Attach()` before the debugger is initialized but that one seems to be harmless.